newIDE/app/src/ExportAndShare/BrowserExporters/BrowserS3PreviewLauncher/index.js

```diff
@@ -219,10 +230,58 @@ export default class BrowserS3PreviewLauncher extends React.Component<
       );
 
       // If the preview windows are new, register them so that they can be accessed
-      // by the debugger.
+      // by the debugger and for the captures to be detected when they close.
       if (!existingPreviewWindow) {
         previewWindows.forEach((previewWindow: WindowProxy) => {
-          registerNewPreviewWindow(previewWindow);
+          const debuggerId = registerNewPreviewWindow(previewWindow);
+          browserPreviewDebuggerServer.registerCallbacks({
```

Comment:
here I'm using the debugger polling in order to know when the window closes and start the preview screenshot upload logic. I don't know if we're ok with that or if I should create a separate polling logic, or rename the debugger one to a more generic "preview"

Reply:
Also, I find very confusing that callbacks are registered as an array of everything, instead of just adding them one by one